proxy/src/main/java/com/velocitypowered/proxy/connection/backend/VelocityServerConnection.java

```diff
@@ -188,6 +189,8 @@ private void startHandshake() {
       handshake.setServerAddress(createBungeeGuardForwardingAddress(secret));
     } else if (proxyPlayer.getConnection().getType() == ConnectionTypes.LEGACY_FORGE) {
       handshake.setServerAddress(playerVhost + HANDSHAKE_HOSTNAME_TOKEN);
+    } else if (proxyPlayer.getConnection().getType() == ConnectionTypes.MODERN_FORGE) {
+      handshake.setServerAddress(playerVhost + MODERN_HANDSHAKE_HOSTNAME_TOKEN);
```

Comment:
Per Forge's implementation, it is legal to add net version after the `FORGE` marker, which should also be forwarded.

https://github.com/MinecraftForge/MinecraftForge/blob/f09fd1f05b39df98f6415b44478b13723cee74cd/src/main/java/net/minecraftforge/network/NetworkContext.java#L96-L110

Reply:
There is currently no need to forward the version. You need to wait for subsequent forge updates before making a decision.

Reply:
It can happen that your implementation forward only `FORGE` when a newer `FORGE2` client connects. Given the facts forge has bump the net version to 3 in prior 1.20.1, this can be considered as bug. You can either reject clients with extra digits after the marker or forward the full content, but not forward part.

Reply:
Given the facts forge has bump the net version to 3 in prior 1.20.1, I prefer implement it properly and save a future pr.

Reply:
Looks like supporting that is pretty much an essential thing, as otherwise it will fail if they ever bump that

Reply:
Completed